### DISCUSSION TITLE
Show instruction count of compiled shader

### DISCUSSION THREAD

**Discussion by priyansh0219** [godotengine/godot-proposals#8248] [General Discussions]
It would be good if godot showed the instruction count of  shader just like in ue5 so we can better optimize it.

**Comment by Calinou** [comment]
When using Vulkan-based renderers, Godot compiles the GLSL shader (which is generated from the Godot shader language) into SPIR-V. This SPIR-V (which is an *intermediate representation*) is then compiled into the binary shader expected by the graphics driver. See RenderingDevice's `shader_compile_spirv_from_source()` and `shader_compile_binary_from_spirv()`).

The question is, at which stage should we report instruction count (and where can we get it easily in a way that can be exposed in the editor)?

Note that the Compatibility rendering method (which uses OpenGL) doesn't use SPIR-V. It goes from GLSL to binary shader directly.
